packages/core/src/Accordion/Accordion.tsx

```diff
@@ -65,18 +65,20 @@ export const HvAccordion = (props: HvAccordionProps) => {
 
   const { classes, cx } = useClasses(classesProp);
 
+  // The isOpen and setIsOpen will only work if the component is uncontrolled
   const [isOpen, setIsOpen] = useControlled(expanded, Boolean(defaultExpanded));
+  const { current: isControlled } = useRef(expanded !== undefined);
 
   const handleAction = useCallback(
     (event: React.SyntheticEvent) => {
       if (!disabled) {
-        onChange?.(event, isOpen);
+        onChange?.(event, isControlled ? !expanded : isOpen);
```

Comment:
I might be missing something, but isn't the problem the value we pass to `onChange`? We should be passing the new value, instead of the current one - right?

```suggestion
        onChange?.(event, !isOpen);
```

Reply:
When I was doing some new tests for the Accordion component, I found out that the expanded state we return to the users through `onChange` isn't accurate when the component is controlled by the user. 
Even though the typical behaviour when controlling the component should be using your own state (like mentioned [here](https://github.com/lumada-design/hv-uikit-react/pull/4180#discussion_r1606925572)), it didn't make sense to me to have an inaccurate expanded value returned by `onChange`.